src/pgduckdb_metadata_cache.cpp

```diff
@@ -62,7 +62,10 @@ InvalidateCaches(Datum arg, int cache_id, uint32 hash_value) {
 		return;
 	}
 	cache.valid = false;
-	list_free(cache.duckdb_only_functions);
+	if (cache.installed) {
+		list_free(cache.duckdb_only_functions);
+		cache.duckdb_only_functions = nullptr;
```

Comment:
nit: for consistency with Postgres code it would be better to set this to NIL (which is defined as `(List *) NULL`)
```suggestion
		cache.duckdb_only_functions = NIL;
```